nbgrader/docs/source/exchange/exchange_plugin.rst

```diff
@@ -0,0 +1,77 @@
+Using an alternative exchange
+=============================
+
+By default, the exchange assumes all users are unique on the host, and uses folders and *file-copys* to manage the distribution of files.
+
+nbgrader will, however, allow alternative mechanisms for distribution.
+
+The default exchange
+--------------------
+
+The exchange package is organised as follows::
+
+    exchange/
+    ├── abc/
+    │   ├── collect.py
+    │   ├── ...
+    │   └── submit.py
+    ├── default/
+    │   ├── collect.py
+    │   ├── ...
+    │   └── submit.py
+    └── exchange_factory.py
+
+The ``exchange.abc`` package contain all the Abstract Base Classes that custom exchange packages need to implement.
```

Comment:
small typo, should say **contains** instead of **contain**

Reply:
Thanks, fixed.